connman/src/service.c

```diff
@@ -9293,14 +9346,19 @@ static void update_from_network(struct connman_service *service,
 
 	if (!service->network) {
 		service->network = connman_network_ref(network);
+		network_update = service->network ? true : false;
 		connman_network_autoconnect_changed(service->network,
 							service->autoconnect);
 		service_schedule_updated(service);
-        }
+	}
 
-	if (was_available != service_available.value(service))
+	if (was_available != service_available.value(service)) {
 		service_boolean_changed(service, &service_available);
 
+		if (network_update)
+			count_available_service_type(service, true);
```

Comment:
Looks like service_available.value(service) is is_available(service) and count_available_service_type(service, true) is nop if is_available(service) == false -> is_available transition true-to-false in here triggers nop-recount and service stays "counted" - is that intentional?

Reply:
If I remember correctly from my tests this actually is intentional as when this is triggered there is an update to the network few lines above (from `NULL` to a `ptr`) so it is assumed that condition changes if also the available value changes.